lib/internal/url.js

```diff
@@ -105,7 +105,11 @@ function parse(input, base) {
       this[context].query = query;
       this[context].fragment = fragment;
       this[context].host = host;
-      this[searchParams] = new URLSearchParams(query);
+      if (this[searchParams]) {  // invoked from href setter
+        initSearchParams(this[searchParams], query);
```

Comment:
I've moved them back. The commits are there for review, may need to squash them before rebase.